File: Projects/iOS/TestLab/ViewController.m
//
//  ViewController.m
//  Testlab
//
//  Created by SalmoJunior on 1/18/16.
//  Copyright © 2016 CI&T. All rights reserved.
//

#import "ViewController.h"
#import "DetailViewController.h"

static NSString *const detail_segue = @"detail_segue";

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UITextField *nameTextField;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    self.title = @"TestLab";
}

 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
     NSString *currentSegueIdentifier = segue.identifier;
     
     if ([currentSegueIdentifier isEqualToString:detail_segue]) {
         DetailViewController *detailViewController = (DetailViewController *)segue.destinationViewController;
         detailViewController.name = self.nameTextField.text;
     }
 }

- (IBAction)next:(id)sender {
    NSString *name = self.nameTextField.text;
    
    if ([name isEqualToString:@""]) {
        [self shakeTextField:self.nameTextField];
    } else {
        [self performSegueWithIdentifier:detail_segue sender:nil];
    }
}

- (void)shakeTextField:(UITextField *)textField
{
    CAKeyframeAnimation * anim = [CAKeyframeAnimation animationWithKeyPath:@"transform"];
    anim.values = @[[NSValue valueWithCATransform3D:CATransform3DMakeTranslation(-5.0f, 0.0f, 0.0f)], [NSValue valueWithCATransform3D:CATransform3DMakeTranslation(5.0f, 0.0f, 0.0f)]];
    anim.autoreverses = YES;
    anim.repeatCount = 2.0f;
    anim.duration = 0.07f;
    
    [textField.layer addAnimation:anim forKey:nil];
}

@end

File: Projects/iOS/TestLab/ViewController.swift
//
//  ViewController.swift
//  TestLab
//
//  Created by SalmoJunior on 1/6/16.
//  Copyright © 2016 CI&T. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var nameTextField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        
        self.title = "TestLab"
    }

    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if (segue.identifier == "detail_segue") {
            let detailViewController = segue.destinationViewController as! DetailViewController
            detailViewController.name = nameTextField.text
        }
        
    }
    
    @IBAction func next(sender: AnyObject) {
        let name = nameTextField.text
        
        if name == "" {
            shakeTextField(nameTextField)
        } else {
            performSegueWithIdentifier("detail_segue", sender: nil)
        }
    }
    
}

func shakeTextField(textField: UITextField) {
    let animation = CABasicAnimation(keyPath: "position")
    animation.duration = 0.07
    animation.repeatCount = 4
    animation.autoreverses = true
    animation.fromValue = NSValue(CGPoint: CGPointMake(textField.center.x - 10, textField.center.y))
    animation.toValue = NSValue(CGPoint: CGPointMake(textField.center.x + 10, textField.center.y))
    textField.layer.addAnimation(animation, forKey: "position")
}

